src/test/functional/features/opening-hours.feature

```diff
@@ -21,15 +21,20 @@ Feature: Opening Hours
     Then the second last opening time is displayed as expected with type shown as selected index 4 and hours as "9:00am to 3:30pm"
     And the last opening time is displayed as expected with type shown as selected index 5 and hours as "10:00am to 4:00pm"
 
-  Scenario: Prevent incomplete entries being added
-    When I enter an incomplete opening hours entry
-    And I click save
-    Then an error message is displayed
-
   Scenario: Remove opening hours
     When I click the remove button under an opening hours entry
     Then I click save
     Then a green update message is displayed in the opening hours tab
     When I click the remove button under an opening hours entry
     Then I click save
     Then a green update message is displayed in the opening hours tab
+
```

Comment:
The new duplicate checking might cause an issue for the existing functional tests. If the functional tests fail after the test where the new opening hours are added and before the test where the new additions are deleted, there is a possibility a duplicate will occur next time the tests are run. This will likely cause the 'add new opening hours' test to fail when the new entry isn't saved due to the duplicate.

Reply:
Currently if the “add opening hours” test fails after adding the opening hours, I thought it still carries on to the “remove opening hours” test? The problem can occur when we only manage to remove one opening hour in the “remove opening hours” test and the test somehow fails after that, resulting in one of the entries not getting removed. The next time we run the “add opening hours” test we might have the problem with duplicates, but the “remove opening hours” test should remove the remaining entry from the first run. So the third run should all be successful. This is not ideal.

We could remove all the entries before starting each test case. We know that the Aberdare County Court has no opening hours, phone number, emails and court types before the functional tests. For the opening hours test, we could add a step at the end of the background step to remove all added entries before starting each test. We can do that in such a way that we count the number of fieldsets before the removal and remove all the present fields (if any), so that the form is empty before starting any new test scenario.

If we do this, we don’t really need the explicit “remove opening hours” scenario as this will be tested as part of the background steps. If we want to keep that scenario, we will need to combine with the ‘add opening hours’ scenario and rename them “Add and remove opening hours” so they are tested together.

Let me know what you think.